src/parser.rs

```diff
@@ -884,6 +946,22 @@ impl Get for Attributes {
     }
 }
 
+fn read_text(parser: &mut Reader) -> Result<Option<String>, Error> {
+    match try!(parser.next()) {
```

Comment:
I realize you had been too tired when fixing this place. ;) Note how every `read_*` function contains a loop not a single read.